test/date_time_parse_test.exs

```diff
@@ -3,5 +3,14 @@ defmodule DateTimeParseTest do
   alias Calendar.DateTime.Parse
   import Parse
   doctest Parse
-end
 
+  test "microsecond precision is capped at 6" do
+    assert(unix!("1000000000.123456789").microsecond == {123457, 6})
```

Comment:
Good catch. It's even worse, since the PR as it stands now would round that to `1000000`, and set `microsecond: {1000000, 6}`. That's not a valid microsecond part. I'll revert back to the truncating behaviour.